### PR TITLE
fix: trade tx

### PR DESCRIPTION
**Pull request by dannythedawger**

### CODE REVIEW

File: v3-sdk/trading/src/config.ts [hunk 1]
```diff
@@ -43,8 +44,8 @@ export const CurrentConfig: ExampleConfig = {
   },
   tokens: {
     in: WETH_TOKEN,
-    amountIn: 0.1,
+    amountIn: 1,
     out: USDC_TOKEN,
-    poolFee: FeeAmount.MEDIUM,
+    poolFee: FeeAmount.LOW,
```
Comment (adjkant) [inline v3-sdk/trading/src/config.ts:49]: Any reason for these changes specifically?

File: v3-sdk/trading/package.json [hunk 2]
```diff
@@ -55,4 +55,4 @@
   "engines": {
     "node": ">=16.0.0"
   }
-}
\ No newline at end of file
+}
```
Comment (adjkant) [inline v3-sdk/trading/package.json]: nit: remove

File: v3-sdk/trading/.eslintrc.json [hunk 3]
```diff
@@ -16,9 +16,12 @@
         "sourceType": "module"
     },
     "plugins": [
-        "typescript-sort-keys"
+        "typescript-sort-keys",
+        "simple-import-sort"
     ],
     "rules": {
+        "simple-import-sort/imports": "error",
+        "simple-import-sort/exports": "error",
```
Comment (adjkant) [inline v3-sdk/trading/.eslintrc.json:24]: Can we add these all at once? I think it makes sense to do another PR for all examples for this if we do it at all but don't think it's critical to include

File: v3-sdk/trading/src/libs/trading.ts [hunk 4]
```diff
@@ -41,7 +55,13 @@ export async function createTrade(): Promise<TokenTrade> {
...
   )
 
-  const amountOut = await getOutputQuote()
+  const provider = getProvider()
+  if (!provider) {
+    throw new Error()
+  }
+
+  const amountOut = await getOutputQuote(swapRoute)
+  console.log('Amount Out', amountOut)
```
Comment (adjkant) [inline v3-sdk/trading/src/libs/trading.ts:64]: remove console log?

File: v3-sdk/trading/src/libs/trading.ts [hunk 5]
```diff
@@ -41,7 +55,13 @@ export async function createTrade(): Promise<TokenTrade> {
     CurrentConfig.tokens.out
   )
 
-  const amountOut = await getOutputQuote()
+  const provider = getProvider()
+  if (!provider) {
+    throw new Error()
+  }
```
Comment (adjkant) [inline v3-sdk/trading/src/libs/trading.ts:61]: run this at the top of the function?

File: v3-sdk/trading/src/libs/trading.ts [hunk 6]
```diff
@@ -50,71 +70,121 @@ export async function createTrade(): Promise<TokenTrade> {
...
     outputAmount: CurrencyAmount.fromRawAmount(
       CurrentConfig.tokens.out,
-      amountOut
+      JSBI.BigInt(amountOut)
     ),
     tradeType: TradeType.EXACT_INPUT,
   })
 
+  console.log('Unchecked Trade', uncheckedTrade)
+
```
Comment (adjkant) [inline v3-sdk/trading/src/libs/trading.ts:83]: remove console log?

File: v3-sdk/trading/src/libs/trading.ts [hunk 7]
```diff
@@ -50,71 +70,121 @@ export async function createTrade(): Promise<TokenTrade> {
...
+  const provider = getProvider()
+
+  if (!walletAddress || !provider) {
     throw new Error('Cannot execute a trade without a connected wallet')
   }
 
   const options: SwapOptions = {
     slippageTolerance: new Percent(500, 10000), // 50 bips, or 0.50%
-    deadline: Math.floor(Date.now() / 1000) + 60 * 20, // 20 minutes from the current Unix time
+    deadline: Math.floor(Date.now() / 1000) + 60 * 30, // 20 minutes from the current Unix time
```
Comment (adjkant) [inline v3-sdk/trading/src/libs/trading.ts:99]: Is this needed? If so we should update comment and maybe other examples

File: v3-sdk/trading/src/libs/trading.ts (hunk 5 above)
Comment (adjkant) [reply v3-sdk/trading/src/libs/trading.ts:61]: I don't think this is actually used actually, remove?

File: v3-sdk/trading/src/libs/trading.ts [hunk 8]
```diff
@@ -50,71 +70,121 @@ export async function createTrade(): Promise<TokenTrade> {
...
+  const tokenApproval = await getTokenTransferApprovals(
+    provider,
+    CurrentConfig.tokens.in.address,
+    walletAddress
+  )
+
+  // Fail if transfer approvals do not go through
+  if (tokenApproval !== TransactionState.Sent) {
+    return TransactionState.Failed
+  }
```
Comment (adjkant) [inline v3-sdk/trading/src/libs/trading.ts:114]: Is this needed needed in addition to the address change?

File: v3-sdk/trading/src/config.ts (hunk 1 above)
Comment (dannydaniil) [reply v3-sdk/trading/src/config.ts:49]: Not necessarily, I can roll back

File: v3-sdk/trading/.eslintrc.json (hunk 3 above)
Comment (dannydaniil) [reply v3-sdk/trading/.eslintrc.json:24]: This example had a bunch of imports and this was really needed. Let's keep it here and I can update the rest too?

File: v3-sdk/trading/src/libs/trading.ts (hunk 8 above)
Comment (dannydaniil) [reply v3-sdk/trading/src/libs/trading.ts:114]: Yes! Required to swap the tokens and probably why the routing guide is failing

File: v3-sdk/trading/src/config.ts [hunk 9]
```diff
@@ -43,7 +44,7 @@ export const CurrentConfig: ExampleConfig = {
   },
   tokens: {
     in: WETH_TOKEN,
-    amountIn: 0.1,
+    amountIn: 1,
```
Comment (adjkant) [inline v3-sdk/trading/src/config.ts:47]: Was this part left in intentionally?

File: v3-sdk/trading/.eslintrc.json (hunk 3 above)
Comment (adjkant) [reply v3-sdk/trading/.eslintrc.json:24]: Okay, let's fast follow PR then please! I just didn't want to require more work to ship this and leaving this off as its own task would make this quicker. I don't think it adds value really still for a user, really only for maintainability.

**Review by adjkant — APPROVED** [review]
Last two + lint fix, otherwise good to go!

**Review by adjkant — APPROVED** [review]
